src/_pytest/python.py

```diff
@@ -1095,7 +1132,7 @@ def _validate_if_using_arg_names(self, argnames, indirect):
                         pytrace=False,
                     )
                 else:
-                    if isinstance(indirect, (tuple, list)):
+                    if isinstance(indirect, Sequence):
```

Comment:
Generalized this a bit to make typing easier. It's also what other functions already check - was inconsistent.